### ISSUE TITLE
MJML type hinting and autocomplete for other file extensions

### ISSUE THREAD

**Issue by S-Braeutigam** [timo-reymann/intellij-mjml-support#199]
### Description

Hi, would it be possible to allow the plugin to support other file extensions than *.mjml? We intend to use it together with laravel blade and when working in a blade.php file we can use mjml components, but the plugin doesn't give us type hinting. We would also be happy to donate a round of coffee if this is a manageable feature request.

**Comment by timo-reymann** [comment]
Currently, the plugin supports [`*_mjml.blade.php`](https://github.com/timo-reymann/intellij-mjml-support/blob/main/src/main/resources/META-INF/php-plugin.xml#L3-L6)

You can set a custom file association tho using as described here: https://www.jetbrains.com/help/idea/creating-and-registering-file-types.html

**Comment by S-Braeutigam** [comment]
Thank you for the quick response. We tried using *_mjml.blade.php as extension and it provides us with autocomplete and type hinting for mjml, but then the Blade autocomplete and type hinting (coming from laravel idea) is not working anymore. Is it possible to setup the mjml plugin in a way that the plugins don't block each other?

**Comment by timo-reymann** [comment]
I will take a look, can you please send me the plugin combination you are using + a simple project where I can test and debug locally? :)

**Comment by S-Braeutigam** [comment]
Sorry for the late reply. We created a small sample project that can be found [here](https://github.com/SOERF/mjml-sample). The README contains a brief summary about the issues.

**Comment by timo-reymann** [comment]
You can always inject other languages (including MJML) into any string fragment in any language, see the docs: https://www.jetbrains.com/help/phpstorm/using-language-injections.html#edit_injected_fragment

**Comment by timo-reymann** [comment]
> Sorry for the late reply. We created a small sample project that can be found [here](https://github.com/SOERF/mjml-sample). The README contains a brief summary about the issues.

Looks awesome, will have a look. Thank you!

**Comment by timo-reymann** [comment]
Okay gotcha, I think I get the problem now. 
I think we have been talking of something else, as mentioned in https://github.com/timo-reymann/intellij-mjml-support/issues/11, were I added the basic association with MJML:

This is due to limitation in the plugin architecture. This does not work well with both MJML & Template language.

So you can either pick associate with MJML and loose the blade highlighting or vice-versa I am afraid. Changing this would basically mean rewriting the plugin, which I don't see value in.

So tl;dr I am sorry but that wont work that way :/

**Comment by S-Braeutigam** [comment]
Alright, thank you for your effort! :)
